src/main/scala/dotty/dokka/tasty/TastyParser.scala

```diff
@@ -32,18 +32,17 @@ case class TastyParser(reflect: Reflection, inspector: DokkaTastyInspector)
     def parseRootTree(root: Tree): Seq[Documentable] = 
       val docs = Seq.newBuilder[Documentable]
       object Traverser extends TreeTraverser:
-        override def traverseTree(tree: Tree)(using ctx: Context): Unit = 
-          tree match {
-            case pck: PackageClause => 
-              docs += parsePackage(pck)
-            case packageObject: ClassDef if(packageObject.symbol.name == "package$") =>
-              docs += parsePackageObject(packageObject)
-            case clazz: ClassDef  =>
-              docs += parseClass(clazz)
-            case _ =>
-          }
-          super.traverseTree(tree)
-
+        override def traverseTree(tree: Tree)(using ctx: Context): Unit = tree match 
+          case pck: PackageClause => 
+            docs += parsePackage(pck)
+            super.traverseTree(tree)
+          case packageObject: ClassDef if(packageObject.symbol.name == "package$") =>
+            docs += parsePackageObject(packageObject)
```

Comment:
One comment here: in Scala 3, package objects are deprecated and we instead have top-level definitions - that is, we can just define a `val`, `var` etc. at the top-level of any file and it will work like it was defined in a Scala 2 package object. I suspect that in the doctool, we will want to display top-level definitions gathered together, much like a package object. I don't think any significant refactoring will be required to do that, but I wanted to let you know.

Reply:
ok, we will need to add support for that. Without package objects, where would users put documentation for package? Some kind of .md file?

Reply:
I think we decided internally on some solution for this, but I'm not sure what was it. I'll ask around.

Reply:
Follow-up: it turns out we just discussed this on the forums, but didn't decide on anything: https://contributors.scala-lang.org/t/dropping-package-objects-and-documentation/2942. I guess that following the Java convention and supporting a `package-info.scala` file is pretty natural, but there's a bunch of things to consider such as do we want to expose this documentation in Tasty-reflect through `.documentation` of the package symbol. I'll discuss this during the next Dotty weekly meeting. Any suggestions you have are welcome.